packages/examples/main/src/langium/main.ts

```diff
@@ -0,0 +1,195 @@
+    const themesUrl = new URL('./resources/themes/themes.json', window.location.href).href;
+    const themeDarkVsUrl = new URL('./resources/themes/theme-defaults~dark_vs.json', window.location.href).href;
+    const themeDarkPlusUrl = new URL('./resources/themes/theme-defaults~dark_plus.json', window.location.href).href;
+
+    const responseLanguageConfig = await fetch(statemachineLanguageConfig);
+    const responseStatemachineTm = await fetch(statemachineTmUrl);
+    const responseOnig = await fetch(onigFileUrl);
+    const responseThemesJson = await fetch(themesUrl);
+    const responseThemeDarkVs = await fetch(themeDarkVsUrl);
+    const responseThemeDarkPlus = await fetch(themeDarkPlusUrl);
+
+    const themes = JSON.parse(await responseThemesJson.text()) as IThemeExtensionPoint[];
+
+    StandaloneServices.initialize({
+        ...getModelEditorServiceOverride(async (model, options) => {
+            console.log('trying to open a model', model, options);
+            return undefined;
+        }),
+        ...getNotificationServiceOverride(),
+        ...getDialogsServiceOverride(),
+        ...getConfigurationServiceOverride(),
+        ...getKeybindingsServiceOverride(),
+        ...getTextmateServiceOverride(async () => {
+            return await responseOnig.arrayBuffer();
+        }),
+        ...getThemeServiceOverride(),
+        ...getTokenClassificationServiceOverride(),
+        ...getLanguageConfigurationServiceOverride(),
+        ...getLanguagesServiceOverride()
+    });
+
+    setDefaultThemes(themes, (theme) => {
+        switch (theme.path) {
+            case './themes/dark_vs.json':
+                return responseThemeDarkVs.text();
+            case './themes/dark_plus.json':
+                return responseThemeDarkPlus.text();
+            default:
+                return Promise.reject(new Error(`Theme ${theme} not found`));
+        }
+    });
+
+    setLanguages([{
+        id: languageId,
+        extensions: [
+            '.statemachine'
+        ],
+        aliases: [
+            'Statemachine',
+            'statemachine'
+        ],
+        configuration: './statemachine-configuration.json'
+    }]);
+    setLanguageConfiguration('/statemachine-configuration.json', () => responseLanguageConfig.text());
+
+    setGrammars([{
+        language: languageId,
+        scopeName: 'source.statemachine',
+        path: './statemachine-grammar.json'
+    }], (grammar) => {
+        switch (grammar.language) {
+            case languageId:
+                return responseStatemachineTm.text();
+            default:
+                return Promise.reject(new Error('Grammar not found!'));
+        }
+    });
+
+    const settingsModel = monaco.editor.createModel(`{
+    "workbench.colorTheme": "Default Dark+",
+    "editor.fontSize": 16,
+    "editor.guides.bracketPairs": true
+}`, 'json', monaco.Uri.parse('inmemory://settings.json'));
+    updateUserConfiguration(settingsModel.getValue());
```

Comment:
Is there a reason to create this model?

Reply:
Is there a better / preferred way to apply configuration? My aim was to set the theme. The other things were just for playing around.

Reply:
you can just do
```typescript
updateUserConfiguration(`{
    "workbench.colorTheme": "Default Dark+",
    "editor.fontSize": 16,
    "editor.guides.bracketPairs": true
}`)
```

btw to set the theme, you can just use `monaco.editor.setTheme`

Reply:
Thanks 👍